testutil/singlevm/setup.sh

```diff
@@ -274,7 +352,92 @@ else
     echo 'dnsmasq command is not supported'
 fi
 
+## Install keystone
+#openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout "$ciao_bin""/keystone_key.pem" -out ${OS_CACERT} -subj "/CN=$ciao_host"
+MYSQL_DATA_DIR="${ciao_bin}"/mysql
+sudo docker run -d -it --name keystone \
+    -p $keystone_int_port:5000 \
+    -p $keystone_ext_port:35357 \
+    -e IDENTITY_HOST="$ciao_host" -e KEYSTONE_ADMIN_PASSWORD="${OS_PASSWORD}" \
+    -v $MYSQL_DATA_DIR:/var/lib/mysql \
+    -v ${OS_CACERT}:/etc/nginx/ssl/keystone_cert.pem \
+    -v ${ciao_pki_path}/${ciao_image_key}:/etc/nginx/ssl/keystone_key.pem clearlinux/keystone
+
+echo -n "Waiting for keystone identity service to become available"
+try_until=$(($(date +%s) + 30))
```

Comment:
I'd suggest a shorter solution, it may require a little change to make it fancier :).

```
until /bin/bash -c "source openrc ; openstack user list"
do
    echo Waiting keystone ...
done
```